hcs-19.md

```diff
@@ -0,0 +1,568 @@
+    "retention_policy": "2_years_default"
+  },
+  "inboundTopicId": "0.0.123456",
+  "outboundTopicId": "0.0.123457"
+}
+```
+
+### Large Data Handling
+
+Where consent‑related documents (e.g., full privacy notice) exceed HCS message
+size, reference the content stored under **HCS‑1**:
+
+```jsonc
+{
+  "p": "hcs-19",
+  "op": "consent_granted",
+  "operator_id": "0.0.123456",
+  "user_id": "user_12345",
+  "consent_data": "hcs://1/0.0.999888",
+  "m": "Large notice stored via HCS‑1"
+}
+```
+
+Referencing the exact notice version satisfies ISO 27560’s requirement to retain
+the presented notice alongside the consent record.
+
+### Retention and Archival
+
+HCS‑19 supports automated retention management:
+
+```jsonc
+{
+  "p": "hcs-19",
+  "op": "retention_check",
+  "operator_id": "0.0.123456",
+  "records_reviewed": 1500,
+  "records_archived": 150,
+  "records_deleted": 25,
+  "retention_policies_applied": ["2_year_default", "1_year_marketing"],
+  "compliance_status": "compliant",
+  "next_review_date": "2025-04-15T10:00:00Z",
+  "m": "Quarterly retention review completed"
+}
+```
+
+## Implementation Workflow
+
+### Step 1: Privacy Topic Setup
+
+1. Create four HCS topics (consent, processing, rights, audit).
+2. Apply memo formats and set submit keys.
+3. Register topic IDs in the agent’s HCS‑11 profile.
+
+### Step 2: Agent Registration
+
+1. Update HCS‑11 profile with privacy‑compliance block.
+2. List supported jurisdictions and standards.
+3. Provide controller and DPO contact details.
+
+### Step 3: User Consent Collection
+
+1. Present a clear privacy notice (ISO 29184 guidance) and collect explicit
+   consent via UI controls.
+2. Log consent events (`consent_granted`, etc.) on the Consent topic.
+3. Verify active consent prior to any data processing.
+4. **Issue a Consent Receipt** – after consent is recorded, provide the user a
+   machine‑ and human‑readable receipt summarising consent ID, controller,
+   purposes, notice reference and timestamp. The receipt includes the HCS
+   transaction ID so users (or auditors) can independently verify authenticity.
+
+### Step 4: Ongoing Compliance Monitoring
+
+1. Record processing events in real time.
+2. Respond to privacy‑rights requests within statutory deadlines.
+3. Conduct and log periodic compliance audits.
+4. Execute automated retention/deletion routines.
+
+## Security Considerations
+
+1. **Data Minimisation** – only include data necessary for compliance.
+2. **Pseudonymisation** – use hashed or tokenised user IDs.
+3. **Access Control & Authenticity** – restrict topic submission to authorised
+   keys and digitally sign messages to guarantee integrity and origin.
+4. **Encryption** – encrypt sensitive payloads before submission.
+5. **Retention Limits** – automatically purge expired records.
+6. **Audit Logging** – maintain a complete, immutable log of compliance
+   activities.
+
+## Examples
+
+### Example 1: User Consent for Customer‑Service Agent
```

Comment:
Not required but recommended, for each example it would be great to have a mermaid diagram of the flow